security/advancedtls/advancedtls.go

```diff
@@ -258,13 +272,18 @@ type ServerOptions struct {
 }
 
 func (o *ClientOptions) config() (*tls.Config, error) {
+	// TODO(gtcooke94) Remove this block when remove o.VerifyPeer
+	// Set AdditionalPeerVerification if the user is still using VerifyPeer.
+	if o.VerifyPeer != nil {
+		o.AdditionalPeerVerification = o.VerifyPeer
```

Comment:
`if o.AdditionalPeerVerification != nil { return <err> }`?

Or ignore the old deprecated field `VerifyPeer` instead by reversing it:

```go
if o.AdditionalPeerVerification == nil {
	o.AdditionalPeerVerification = o.VerifyPeer
}
```

Reply:
It's not required, so the error return I think doesn't make the most sense.

The reverse works too, I guess it just matters for the precedence - I was going with "if the old field is set they probably haven't migrated", but the bottom is good too as a "if the new field isn't set take whatever is in the old field"

Reply:
The error would be "you set two fields that mean the same thing, one of which is deprecated: what were you thinking?" :laughing: